src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1235,6 +1242,224 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetupSrpHost(co
     return error;
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsResponseToMdnsData(otDnsServiceInfo & serviceInfo,
+                                                                                      char * serviceType,
+                                                                                      chip::Mdns::MdnsService & mdnsService,
+                                                                                      DnsServiceTxtEntries & serviceTxtEntries)
+{
+    char protocol[chip::Mdns::kMdnsProtocolTextMaxSize + 1];
+
+    // Extract from the <hostname>.default.service.arpa. the <hostname> part.
+    size_t substringSize = strlen(serviceInfo.mHostNameBuffer) - 1 - chip::Mdns::kMdnsDefaultSrpDomainSize - 1;
```

Comment:
I think it's unsafe to assume that the domain name is default.service.arpa. One can change it with otSrpClientSetDomainName. I think you should a helper function like: `CopyWithoutDomainName(out, outLen, in)` which would drop the domain name without no matter what it is.